docs/docs/guides/ORIENTATION.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: orientation
+title: Orientation
+sidebar_label: Orientation
+---
+
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+import useBaseUrl from '@docusaurus/useBaseUrl'
+
+<div class="image-container">
+  <img width="283" src={useBaseUrl("img/example.png")} />
+</div>
+
+## Camera Sensor Orientation
+
+A Camera sensor is configured to deliver buffers in a specific size, for example 4096x2160.
+To avoid re-allocating such large buffers every time the phone rotates, the Camera pipeline will always deliver buffers in it's native sensor orientation (see [`sensorOrientation`](/docs/api/interfaces/CameraDevice#sensororientation)) and frames need to be rotated to appear up-right.
+
+Since actually rotating pixels in such large buffers is really expensive and causes an unnecessary performance overhead, VisionCamera applies rotations through flags or transforms:
+
+#### Photo & Video Output
+
+Photos and videos will be captured in a potentially "wrong" orientation, and VisionCamera will write an EXIF flag to the photo/video file with the correct presentation rotation.
+
+:::info
+This is handled automatically, and it's behaviour can be controlled via the [`outputOrientation`](/docs/api/interfaces/CameraProps#outputorientation) property.
+:::
+
+#### Frame Processor Output
+
+Frame Processors will stream frames in a potentially "wrong" orientation, and the client is responsible for properly interpreting the Frame data.
+
+:::info
+This needs to be handled manually, see [`Frame.orientation`](/docs/api/interfaces/Frame#orientation).
+For example, in MLKit just pass the `Frame`'s `orientation` to the `detect(...)` method.
+:::
+
+#### Preview View
+
+The Preview output will stream in a potentially "wrong" orientation, but uses view transforms (rotate + translate matrix) to properly display the Camera stream "up-right".
+
+:::info
+This will always happen automatically according to the screen's rotation.
+:::
+
+### The `outputOrientation` prop
+
+The orientation in which photos and videos are captured can be adjusted via the [`outputOrientation`](/docs/api/interfaces/CameraProps#outputorientation) property:
+
+```tsx
+<Camera {...props} outputOrientation="device" />
```

Comment:
Would it make sense to combine these two properties into a single object, and pass it as config?

So it will look as:

```tsx
const orientation = {
  sensor: "...",
  output: "...",
};

// ...

<Camera orientation={orientation} />
```

Asking from curiosity 🙂

Reply:
Hey Kiryll - not really - sensor orientation is something fixed by the Camera (it is relative to the phone's natural orientation), and the `outputOrientation` just specifies where to rotate towards (either follow `device`/`preview` orientation, or fixed to something like `landscape-left`).

Hope that makes sense :)

Reply:
@mrousavy ah, yeah, makes sense then 🙂